src/js/components/AppListComponent.jsx

```diff
@@ -80,12 +113,42 @@ var AppListComponent = React.createClass({
     });
   },
 
+  getAppsInGroup: function () {
+    var apps = this.state.apps;
+    var currentGroup = this.props.currentGroup;
+
+    return lazy(apps)
+      .filter((app) => app.id.startsWith(currentGroup))
+      .reduce((memo, app) => {
```

Comment:
All fair points.

1. Agree!
2. "groups" do not exist as entities (only apps) so we need to "create" them when in the list view. I also struggled a bit with the naming for `createGroup`, but to be honest that's exactly what we are doing.  
3. Leftover. The idea was to get the folders always at the top. But obviously, as you noticed, that doesn't work. I'll make it a `push()` and pospone that discussion for now

Thanks!